scripts/Island.gd

```diff
@@ -0,0 +1,20 @@
+tool
+class_name Island
+extends Spatial
+
+const TREES: Array = [
+	preload("res://scenes/Tree1.tscn"),
+	preload("res://scenes/Tree2.tscn"),
+	preload("res://scenes/Maple.tscn")
+]
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	var i: int = 0
+	for child in get_children():
+		if child is Spatial:
+			if child.name.begins_with("tree"):
+				var child_tree = TREES[i % TREES.size()].instance()
```

Comment:
OK so does this mean you are looking through the island .gltf for any objects named with "tree" and telling the engine to draw a tree there?

Is this how you would bring in all the objects - rocks, houses, flowers, etc. or does this get computationally expensive with all the for loops?

Reply:
Rocks and flowers should generally probably be scattered procedurally rather than being predefined as these trees are; that would happen in a separate function.

Houses are more interesting, and should actually be regular scenes manually placed in the Island in the IDE